pkg/controllers/jobset_controller.go

```diff
@@ -238,10 +239,17 @@ func (r *JobSetReconciler) suspendJobSet(ctx context.Context, js *jobset.JobSet,
 }
 
 func (r *JobSetReconciler) resumeJobSetIfNecessary(ctx context.Context, js *jobset.JobSet, ownedJobs *childJobs) error {
+	nodeAffinities := map[string]map[string]string{}
+	for _, replicatedJob := range js.Spec.ReplicatedJobs {
+		nodeAffinities[fmt.Sprintf("%s-%s", js.Name, replicatedJob.Name)] = replicatedJob.Template.Spec.Template.Spec.NodeSelector
+	}
```

Comment:
just use replicatedJob.Name as the map key; each job has a label with the replicatedJob name under the key jobset.ReplicatedJobNameKey